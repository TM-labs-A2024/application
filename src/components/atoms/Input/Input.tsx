import { Input } from '@chakra-ui/react'
import React from 'react'

export default function InputComponent({
  placeholder,
  label,
  className,
  disabled
}: {
  placeholder: string
  label: string
  className: string
  disabled: boolean
}) {
  return (
    <div className={className}>
      <label>{label}</label>
      <Input placeholder={placeholder} size="md" disabled={disabled} />
    </div>
  )
}
