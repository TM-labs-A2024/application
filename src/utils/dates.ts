export const formatDate = (date: string) => {
  const chunks = date.split('-')

  const newDate = chunks[1] + '-' + chunks[2] + '-' + chunks[0]

  return newDate
}
