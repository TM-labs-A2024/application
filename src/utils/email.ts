import emailjs from '@emailjs/browser'

export function sendEmail(data: {
  from_name: string
  to_name: string
  code: string | number
  to_email: string
}) {
  const templateParams = {
    ...data
  }

  const serviceKey = process.env.NEXT_PUBLIC_EMAIL_SERVICE_KEY ?? ''
  const templateKey = process.env.NEXT_PUBLIC_EMAIL_TEMPLATE_KEY ?? ''
  const apiKey = process.env.NEXT_PUBLIC_EMAIL_API_KEY ?? ''

  emailjs.send(serviceKey, templateKey, templateParams, apiKey).catch(() => {
    alert('Algo sali√≥ mal, por favor reintenta el registro.')
  })
}
