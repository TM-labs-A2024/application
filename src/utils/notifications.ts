import { GENERIC_ERROR } from '@constants/index'
import { isMobile } from '@utils/index'
import { Store } from 'react-notifications-component'

export const setupErrorNotification = () => {
  Store.addNotification(
    GENERIC_ERROR('Algo salió mal, por favor inténtelo de nuevo.', isMobile(window))
  )
}
