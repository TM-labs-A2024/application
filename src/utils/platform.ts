import { Capacitor } from '@capacitor/core'

export const isWeb = () => Capacitor.getPlatform() === 'web'

export const isIOS = () => Capacitor.getPlatform() === 'ios'

export const isAndroid = () => Capacitor.getPlatform() === 'android'
