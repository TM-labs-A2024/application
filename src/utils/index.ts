import { formatDate } from './dates'
import { sendEmail, generateVerificationCode, setupEmailSending } from './email'
import { setupErrorNotification } from './notifications'
import { isWeb, isIOS, isAndroid, isMobile } from './platform'

export {
  isWeb,
  isIOS,
  isAndroid,
  isMobile,
  sendEmail,
  generateVerificationCode,
  setupEmailSending,
  setupErrorNotification,
  formatDate
}
