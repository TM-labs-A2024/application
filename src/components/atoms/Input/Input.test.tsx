import { render, screen, fireEvent } from '@test/utils/index'

import '@testing-library/jest-dom'
import Input from './Input'

const props = {
  placeholder: 'Placeholder text',
  label: 'Label',
  className: '',
  disabled: false,
  id: ''
}

const disabledProps = {
  ...props,
  disabled: true
}

describe('Atoms > Input test', () => {
  test('The component shows the expected label', async () => {
    render(<Input {...props} />)

    expect(await screen.findByText(props.label)).toBeInTheDocument()
  })

  test("The component shows the placeholder text when it's empty", () => {
    render(<Input {...props} />)

    expect(screen.getByPlaceholderText(props.placeholder)).toBeInTheDocument()
  })

  test('The component shows the written text when receiving input', () => {
    render(<Input {...props} />)

    const input = screen.getByPlaceholderText(props.placeholder) as HTMLInputElement
    fireEvent.change(input, { target: { value: 'Test text' } })

    expect(input.value).toBe('Test text')
  })

  test('The component is disabled when the disabled prop is true', async () => {
    render(<Input {...disabledProps} />)

    expect(screen.getByPlaceholderText(disabledProps.placeholder)).toBeDisabled()
  })

  test('Matches the snapshot', async () => {
    render(<Input {...props} />)

    expect(await screen.findByTestId('input-component')).toMatchSnapshot()
  })
})
