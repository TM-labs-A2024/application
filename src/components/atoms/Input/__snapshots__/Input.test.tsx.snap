// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms > Input test Matches the snapshot 1`] = `
<div
  class=""
  data-testid="input-component"
>
  <label>
    Label
  </label>
  <input
    class="chakra-input min-h-10 css-1cjy4zv"
    id=""
    placeholder="Placeholder text"
  />
</div>
`
